File: backend/src/Sgr/V1/ReciboBundle/Entity/Parametro.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

namespace App\Sgr\V1\ReciboBundle\Entity;

/**
 * Description of UsuarioGrupo
 *
 * @author odenilton
 */
use Doctrine\ORM\Mapping as ORM;
use App\Sgr\BaseBundle\Utils\Configurator;

/**
 * @ORM\Entity
 * @ORM\Table(name="parametro")
 * @ORM\Entity(repositoryClass="App\Sgr\V1\ReciboBundle\Repository\ParametroRepository")
 */
class Parametro {

    /**
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue
     * @var int
     */
    private $id;

    /**
     * @ORM\Column(name="descricao", type="string")
     * @var string
     */
    private $descricao;

    /**
     * @ORM\Column(name="mes", type="string")
     * @var string
     */
    private $mes;

    /**
     * @ORM\Column(name="ano", type="string")
     * @var string
     */
    private $ano;

    /**
     * @ORM\Column(name="impostoISS", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $impostoISS;

    /**
     * @ORM\Column(name="inssFaixa1", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $inssFaixa1;

    /**
     * @ORM\Column(name="inssFaixa2", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $inssFaixa2;

    /**
     * @ORM\Column(name="inssFaixa3", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $inssFaixa3;

    /**
     * @ORM\Column(name="inssFaixa4", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $inssFaixa4;

    /**
     * @ORM\Column(name="inssIndice1", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $inssIndice1;

    /**
     * @ORM\Column(name="inssIndice2", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $inssIndice2;

    /**
     * @ORM\Column(name="inssIndice3", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $inssIndice3;

    /**
     * @ORM\Column(name="inssIndice4", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $inssIndice4;

    /**
     * @ORM\Column(name="irDeducao", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irDeducao;

    /**
     * @ORM\Column(name="irDeducao65", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irDeducao65;

    /**
     * @ORM\Column(name="irDeducaoPrevPro", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irDeducaoPrevPro;

    /**
     * @ORM\Column(name="irBase1", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irBase1;

    /**
     * @ORM\Column(name="irBase2", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irBase2;

    /**
     * @ORM\Column(name="irBase3", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irBase3;

    /**
     * @ORM\Column(name="irBase4", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irBase4;

    /**
     * @ORM\Column(name="irPercentual1", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irPercentual1;

    /**
     * @ORM\Column(name="irPercentual2", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irPercentual2;

    /**
     * @ORM\Column(name="irPercentual3", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irPercentual3;

    /**
     * @ORM\Column(name="irPercentual4", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irPercentual4;

    /**
     * @ORM\Column(name="irParcela1", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irParcela1;

    /**
     * @ORM\Column(name="irParcela2", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irParcela2;

    /**
     * @ORM\Column(name="irParcela3", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irParcela3;

    /**
     * @ORM\Column(name="irParcela4", type="decimal", nullable=true, precision=10, scale=2)
     * @var decimal
     */
    private $irParcela4;

    /**
     * @ORM\ManyToOne(targetEntity="App\Sgr\V1\ReciboBundle\Entity\Orgao")
     * @ORM\JoinColumn(name="orgao_id", referencedColumnName="id")
     */
    protected $orgao;

    public function __construct($options = null) {
        Configurator::configure($this, $options);
    }

    function getId() {
        return $this->id;
    }

    function getDescricao() {
        return $this->descricao;
    }

    function getMes() {
        return $this->mes;
    }

    function getAno() {
        return $this->ano;
    }

    function getImpostoISS() {
        return $this->impostoISS;
    }

    function getInssFaixa1() {
        return $this->inssFaixa1;
    }

    function getInssFaixa2() {
        return $this->inssFaixa2;
    }

    function getInssFaixa3() {
        return $this->inssFaixa3;
    }

    function getInssFaixa4() {
        return $this->inssFaixa4;
    }

    function getInssIndice1() {
        return $this->inssIndice1;
    }

    function getInssIndice2() {
        return $this->inssIndice2;
    }

    function getInssIndice3() {
        return $this->inssIndice3;
    }

    function getInssIndice4() {
        return $this->inssIndice4;
    }

    function getIrDeducao() {
        return $this->irDeducao;
    }

    function getIrDeducao65() {
        return $this->irDeducao65;
    }

    function getIrDeducaoPrevPro() {
        return $this->irDeducaoPrevPro;
    }

    function getIrBase1() {
        return $this->irBase1;
    }

    function getIrBase2() {
        return $this->irBase2;
    }

    function getIrBase3() {
        return $this->irBase3;
    }

    function getIrBase4() {
        return $this->irBase4;
    }

    function getIrPercentual1() {
        return $this->irPercentual1;
    }

    function getIrPercentual2() {
        return $this->irPercentual2;
    }

    function getIrPercentual3() {
        return $this->irPercentual3;
    }

    function getIrPercentual4() {
        return $this->irPercentual4;
    }

    function getIrParcela1() {
        return $this->irParcela1;
    }

    function getIrParcela2() {
        return $this->irParcela2;
    }

    function getIrParcela3() {
        return $this->irParcela3;
    }

    function getIrParcela4() {
        return $this->irParcela4;
    }

    function getOrgao() {
        return $this->orgao;
    }

    function setId($id) {
        $this->id = $id;
        return $this;
    }

    function setDescricao($descricao) {
        $this->descricao = $descricao;
        return $this;
    }

    function setMes($mes) {
        $this->mes = $mes;
        return $this;
    }

    function setAno($ano) {
        $this->ano = $ano;
        return $this;
    }

    function setImpostoISS($impostoISS) {
        $this->impostoISS = $impostoISS;
        return $this;
    }

    function setInssFaixa1($inssFaixa1) {
        $this->inssFaixa1 = $inssFaixa1;
        return $this;
    }

    function setInssFaixa2($inssFaixa2) {
        $this->inssFaixa2 = $inssFaixa2;
        return $this;
    }

    function setInssFaixa3($inssFaixa3) {
        $this->inssFaixa3 = $inssFaixa3;
        return $this;
    }

    function setInssFaixa4($inssFaixa4) {
        $this->inssFaixa4 = $inssFaixa4;
        return $this;
    }

    function setInssIndice1($inssIndice1) {
        $this->inssIndice1 = $inssIndice1;
        return $this;
    }

    function setInssIndice2($inssIndice2) {
        $this->inssIndice2 = $inssIndice2;
        return $this;
    }

    function setInssIndice3($inssIndice3) {
        $this->inssIndice3 = $inssIndice3;
        return $this;
    }

    function setInssIndice4($inssIndice4) {
        $this->inssIndice4 = $inssIndice4;
        return $this;
    }

    function setIrDeducao($irDeducao) {
        $this->irDeducao = $irDeducao;
        return $this;
    }

    function setIrDeducao65($irDeducao65) {
        $this->irDeducao65 = $irDeducao65;
        return $this;
    }

    function setIrDeducaoPrevPro($irDeducaoPrevPro) {
        $this->irDeducaoPrevPro = $irDeducaoPrevPro;
        return $this;
    }

    function setIrBase1($irBase1) {
        $this->irBase1 = $irBase1;
        return $this;
    }

    function setIrBase2($irBase2) {
        $this->irBase2 = $irBase2;
        return $this;
    }

    function setIrBase3($irBase3) {
        $this->irBase3 = $irBase3;
        return $this;
    }

    function setIrBase4($irBase4) {
        $this->irBase4 = $irBase4;
        return $this;
    }

    function setIrPercentual1($irPercentual1) {
        $this->irPercentual1 = $irPercentual1;
        return $this;
    }

    function setIrPercentual2($irPercentual2) {
        $this->irPercentual2 = $irPercentual2;
        return $this;
    }

    function setIrPercentual3($irPercentual3) {
        $this->irPercentual3 = $irPercentual3;
        return $this;
    }

    function setIrPercentual4($irPercentual4) {
        $this->irPercentual4 = $irPercentual4;
        return $this;
    }

    function setIrParcela1($irParcela1) {
        $this->irParcela1 = $irParcela1;
        return $this;
    }

    function setIrParcela2($irParcela2) {
        $this->irParcela2 = $irParcela2;
        return $this;
    }

    function setIrParcela3($irParcela3) {
        $this->irParcela3 = $irParcela3;
        return $this;
    }

    function setIrParcela4($irParcela4) {
        $this->irParcela4 = $irParcela4;
        return $this;
    }

    function setOrgao($orgao) {
        $this->orgao = $orgao;
        return $this;
    }

}

File: backend-java/src/main/java/br/com/oljsoft/sgr/model/Parametro.java
package br.com.oljsoft.sgr.model;

import java.io.Serializable;
import java.math.BigDecimal;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Table(name = "tab_parametro")
public class Parametro implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	@Column(name = "param_descricao")
	private String descricao;

	@Column(name = "param_mes")
	private String mes;

	@Column(name = "param_ano")
	private String ano;
	
	@Column(name = "param_imposto_ISS")
	private Number impostoISS;

	@Column(name = "param_inss_faixa1")
	private BigDecimal inssFaixa1;

	@Column(name = "param_inss_faixa2")
	private BigDecimal inssFaixa2;

	@Column(name = "param_inss_faixa3")
	private BigDecimal inssFaixa3;

	@Column(name = "param_inss_faixa4")
	private BigDecimal inssFaixa4;

	@Column(name = "param_inss_indice1")
	private Number inssIndice1;

	@Column(name = "param_inss_indice2")
	private Number inssIndice2;

	@Column(name = "param_inss_indice3")
	private Number inssIndice3;

	@Column(name = "param_inss_indice4")
	private Number inssIndice4;

	@Column(name = "param_ir_deducao")
	private BigDecimal irDeducao;

	@Column(name = "param_ir_deducao65")
	private BigDecimal irDeducao65;

	@Column(name = "param_ir_deducao_prev_prop")
	private BigDecimal irDeducaoPrevPro;

	@Column(name = "param_ir_base1")
	private Number irBase1;

	@Column(name = "param_ir_base2")
	private Number irBase2;

	@Column(name = "param_ir_base3")
	private Number irBase3;

	@Column(name = "param_ir_base4")
	private Number irBase4;

	@Column(name = "param_ir_percentual1")
	private Number irPercentual1;

	@Column(name = "param_ir_percentual2")
	private Number irPercentual2;

	@Column(name = "param_ir_percentual3")
	private Number irPercentual3;

	@Column(name = "param_ir_percentual4")
	private Number irPercentual4;

	@Column(name = "param_ir_parcela1")
	private Number irParcela1;

	@Column(name = "param_ir_parcela2")
	private Number irParcela2;

	@Column(name = "param_ir_parcela3")
	private Number irParcela3;

	@Column(name = "param_ir_parcela4")
	private Number irParcela4;
	
	@ManyToOne
	@JoinColumn(name = "param_id_orgao")
	@JsonBackReference(value = "parametros")
	private Orgao orgao;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Number getImpostoISS() {
		return impostoISS;
	}

	public void setImpostoISS(Number impostoISS) {
		this.impostoISS = impostoISS;
	}

	public String getDescricao() {
		return descricao;
	}

	public Orgao getOrgao() {
		return orgao;
	}

	public void setOrgao(Orgao orgao) {
		this.orgao = orgao;
	}

	public void setDescricao(String descricao) {
		this.descricao = descricao;
	}

	public String getMes() {
		return mes;
	}

	public void setMes(String mes) {
		this.mes = mes;
	}

	public String getAno() {
		return ano;
	}

	public void setAno(String ano) {
		this.ano = ano;
	}

	public BigDecimal getInssFaixa1() {
		return inssFaixa1;
	}

	public void setInssFaixa1(BigDecimal inssFaixa1) {
		this.inssFaixa1 = inssFaixa1;
	}

	public BigDecimal getInssFaixa2() {
		return inssFaixa2;
	}

	public void setInssFaixa2(BigDecimal inssFaixa2) {
		this.inssFaixa2 = inssFaixa2;
	}

	public BigDecimal getInssFaixa3() {
		return inssFaixa3;
	}

	public void setInssFaixa3(BigDecimal inssFaixa3) {
		this.inssFaixa3 = inssFaixa3;
	}

	public BigDecimal getInssFaixa4() {
		return inssFaixa4;
	}

	public void setInssFaixa4(BigDecimal inssFaixa4) {
		this.inssFaixa4 = inssFaixa4;
	}

	public Number getInssIndice1() {
		return inssIndice1;
	}

	public void setInssIndice1(Number inssIndice1) {
		this.inssIndice1 = inssIndice1;
	}

	public Number getInssIndice2() {
		return inssIndice2;
	}

	public void setInssIndice2(Number inssIndice2) {
		this.inssIndice2 = inssIndice2;
	}

	public Number getInssIndice3() {
		return inssIndice3;
	}

	public void setInssIndice3(Number inssIndice3) {
		this.inssIndice3 = inssIndice3;
	}

	public Number getInssIndice4() {
		return inssIndice4;
	}

	public void setInssIndice4(Number inssIndice4) {
		this.inssIndice4 = inssIndice4;
	}

	public BigDecimal getIrDeducao() {
		return irDeducao;
	}

	public void setIrDeducao(BigDecimal irDeducao) {
		this.irDeducao = irDeducao;
	}

	public BigDecimal getIrDeducao65() {
		return irDeducao65;
	}

	public void setIrDeducao65(BigDecimal irDeducao65) {
		this.irDeducao65 = irDeducao65;
	}

	public BigDecimal getIrDeducaoPrevPro() {
		return irDeducaoPrevPro;
	}

	public void setIrDeducaoPrevPro(BigDecimal irDeducaoPrevPro) {
		this.irDeducaoPrevPro = irDeducaoPrevPro;
	}

	public Number getIrBase1() {
		return irBase1;
	}

	public void setIrBase1(Number irBase1) {
		this.irBase1 = irBase1;
	}

	public Number getIrBase2() {
		return irBase2;
	}

	public void setIrBase2(Number irBase2) {
		this.irBase2 = irBase2;
	}

	public Number getIrBase3() {
		return irBase3;
	}

	public void setIrBase3(Number irBase3) {
		this.irBase3 = irBase3;
	}

	public Number getIrBase4() {
		return irBase4;
	}

	public void setIrBase4(Number irBase4) {
		this.irBase4 = irBase4;
	}

	public Number getIrPercentual1() {
		return irPercentual1;
	}

	public void setIrPercentual1(Number irPercentual1) {
		this.irPercentual1 = irPercentual1;
	}

	public Number getIrPercentual2() {
		return irPercentual2;
	}

	public void setIrPercentual2(Number irPercentual2) {
		this.irPercentual2 = irPercentual2;
	}

	public Number getIrPercentual3() {
		return irPercentual3;
	}

	public void setIrPercentual3(Number irPercentual3) {
		this.irPercentual3 = irPercentual3;
	}

	public Number getIrPercentual4() {
		return irPercentual4;
	}

	public void setIrPercentual4(Number irPercentual4) {
		this.irPercentual4 = irPercentual4;
	}

	public Number getIrParcela1() {
		return irParcela1;
	}

	public void setIrParcela1(Number irParcela1) {
		this.irParcela1 = irParcela1;
	}

	public Number getIrParcela2() {
		return irParcela2;
	}

	public void setIrParcela2(Number irParcela2) {
		this.irParcela2 = irParcela2;
	}

	public Number getIrParcela3() {
		return irParcela3;
	}

	public void setIrParcela3(Number irParcela3) {
		this.irParcela3 = irParcela3;
	}

	public Number getIrParcela4() {
		return irParcela4;
	}

	public void setIrParcela4(Number irParcela4) {
		this.irParcela4 = irParcela4;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Parametro other = (Parametro) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

}
